### PR TITLE
CNV-1617: Clone VM

### PR DESCRIPTION
**Pull request by ousleyp**
Modularizing clone VM/disk procedures for CNV 2.0.

Jira: https://jira.coreos.com/browse/CNV-1617

### CODE REVIEW

File: modules/cnv-cloning-pvc-of-vm-disk-into-new-datavolume.adoc [hunk 1]
```diff
@@ -0,0 +1,69 @@
+// Module included in the following assemblies:
+//
+// * cnv/cnv_users_guide/cnv-cloning-vm-disk-into-new-datavolume.adoc
+
+[id="cnv-cloning-pvc-of-vm-disk-into-new-datavolume_{context}"]
+= Cloning the PersistentVolumeClaim of a virtual machine disk into a new DataVolume
+
+You can clone a PVC of an existing virtual machine disk into a new DataVolume.
```
Comment (kalexand-rh) [inline modules/cnv-cloning-pvc-of-vm-disk-into-new-datavolume.adoc]: PersistentVolumeClaim (PVC)

File: modules/cnv-cloning-pvc-of-vm-disk-into-new-datavolume.adoc [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+The new DataVolume can then be used for a new virtual machine.
+
+[NOTE]
+====
+When a DataVolume is created independently of a virtual machine, the lifecycle
+of the DataVolume is independent of the virtual machine. If the virtual machine
+is deleted, neither the DataVolume nor its associated PVC will be deleted.
```
Comment (kalexand-rh) [inline modules/cnv-cloning-pvc-of-vm-disk-into-new-datavolume.adoc]: s/will be/is

File: modules/cnv-cloning-pvc-of-vm-disk-into-new-datavolume.adoc [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+====
+
+.Prerequisites
+* A PVC of an existing virtual machine disk. The associated virtual machine must
```
Comment (kalexand-rh) [inline modules/cnv-cloning-pvc-of-vm-disk-into-new-datavolume.adoc]: This list isn't parallel. Maybe "Determine the PVC of an existing virtual machine disk to use. You must power down the virtual machine that is associated with the PVC before you can clone it."

File: modules/cnv-cloning-pvc-of-vm-disk-into-new-datavolume.adoc [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+----
+$ oc create -f <cloner-datavolume>.yaml
+----
++
+[NOTE]
+====
+DataVolumes prevent a virtual machine from starting before the PVC is prepared,
+so you can create a virtual machine that references the new DataVolume while the
+PVC clones.
+====
```
Comment (kalexand-rh) [inline modules/cnv-cloning-pvc-of-vm-disk-into-new-datavolume.adoc]: Is there a good next step for this procedure?

File: modules/cnv-creating-new-vm-from-cloned-pvc-using-datavolumetemplate.adoc [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+          name: "my-favorite-vm-disk"
+----
+<1> The virtual machine to create.
+
+. Create the virtual machine with the PVC-cloned DataVolume:
++
+----
+$ oc create -f <vm-clone-datavolumetemplate>.yaml
+----
+
```
Comment (kalexand-rh) [inline modules/cnv-creating-new-vm-from-cloned-pvc-using-datavolumetemplate.adoc]: Is there a next step for this procedure?

File: cnv/cnv_users_guide/cnv-cloning-vm-disk-into-new-datavolume.adoc [hunk 6]
```diff
@@ -0,0 +1,15 @@
+[id="cnv-cloning-vm-disk-into-new-datavolume"]
+= Cloning a virtual machine disk into a new DataVolume
+include::modules/cnv-document-attributes.adoc[]
+:context: cnv-cloning-vm-disk-into-new-datavolume
+toc::[]
+
+You can clone the PersistentVolumeClaim (PVC) of a virtual machine disk into
+a new DataVolume by referencing the source PVC in your DataVolume configuration
+file.
```
Comment (kalexand-rh) [inline cnv/cnv_users_guide/cnv-cloning-vm-disk-into-new-datavolume.adoc]: Is there a good reason why a user might want to do one of these over the other? They seem similar enough that some clarity about why you'd pick one over the other might be helpful in both assemblies, if you have it.

**Review by kalexand-rh — COMMENTED** [review]
Couple of questions and picks, but this change is looking good.

File: modules/cnv-creating-new-vm-from-cloned-pvc-using-datavolumetemplate.adoc [hunk 7]
```diff
@@ -0,0 +1,89 @@
...
+an existing virtual machine into a DataVolume. By referencing a
+`dataVolumeTemplate` in the virtual machine `spec`, the `source` PVC is cloned to
+a DataVolume, which is then automatically used for the creation of the virtual
+machine.
+
+[NOTE]
+====
+When a DataVolume is created as part of the DataVolumeTemplate of a virtual
+machine, the lifecycle of the DataVolume is then dependent on the virtual
+machine: If the virtual machine is deleted, the DataVolume and associated
```
[the remark this answers is not in the document]
Comment (ousleyp) [reply modules/cnv-creating-new-vm-from-cloned-pvc-using-datavolumetemplate.adoc]: I'd already done that, luckily. :) You must've caught it right before that commit.

[the remark this answers is not in the document]
Comment (ousleyp) [reply modules/cnv-creating-new-vm-from-cloned-pvc-using-datavolumetemplate.adoc]: I *thought* I'd already fixed that, but it was a slightly different note. Fixing now.

File: modules/cnv-cloning-pvc-of-vm-disk-into-new-datavolume.adoc (hunk 4 above)
Comment (ousleyp) [reply modules/cnv-cloning-pvc-of-vm-disk-into-new-datavolume.adoc]: I'm not sure. It's optional to create a VM at that point, but that's the next step for some customers.

File: modules/cnv-creating-new-vm-from-cloned-pvc-using-datavolumetemplate.adoc (hunk 5 above)
Comment (ousleyp) [reply modules/cnv-creating-new-vm-from-cloned-pvc-using-datavolumetemplate.adoc]: I don't think so. The main part of that procedure is just editing the config file.

File: cnv/cnv_users_guide/cnv-cloning-vm-disk-into-new-datavolume.adoc (hunk 6 above)
Comment (ousleyp) [reply cnv/cnv_users_guide/cnv-cloning-vm-disk-into-new-datavolume.adoc]: Maybe some specific use cases could be added later, but I don't think I have that info yet. It boils down to deciding whether or not to immediately create a VM.

Comment (aglitke) [reply cnv/cnv_users_guide/cnv-cloning-vm-disk-into-new-datavolume.adoc]: In one example you might first create a VM and prepare it (by updating, configuring, installing software, etc).  Then you may clone the disk to a new DataVolume to serve as the basis for a VM template.  In this case you would not require (nor want) a VM associated with that disk.  At a later time, you could create multiple VMs by cloning that disk using the second method (DataVolume template inside a VM object).  Hopefully this clears it up a little.

**Pull request merged**